src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -152,14 +153,21 @@ static MultiTable * MultiSubqueryPushdownTable(Query *subquery);
  * plan and adds a root node to top of it. The  original query is only used for subquery
  * pushdown planning.
  *
+ * In order to support external parameters for the queries where planning
+ * is done on the original query, we need to replace the external parameters
+ * manually. To achive that for subquery pushdown planning, we pass boundParams
+ * to this function. We need to do that since Citus currently unable to send
+ * parameters to the workers on the execution.
+ *
  * We also pass queryTree and plannerRestrictionContext to the planner. They
  * are primarily used to decide whether the subquery is safe to pushdown.
  * If not, it helps to produce meaningful error messages for subquery
  * pushdown planning.
  */
 MultiTreeRoot *
 MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
```

Comment:
> As a side question, can't we implement the prepared statement support in general as we do it here? (i..e, Move ResolveExternalParams() before standard_planner() call?)

No, the parameters aren't generally available, only when a "custom" plan is chosen, not when a generic one is chosen.   In the cases where they are, that's pretty much how it already works, except it's done by standard_planner().